### PR TITLE
[occm] Fix Naming of listener, pools and monitors during Creation

### LINKED ISSUE #1965: [occm] Wrong Naming for new created listener, pools and monitors for load balancer
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

During the creation of a new load balancer the names for listener, pools and monitors are either not set at all or set differently to later during updates.

**What you expected to happen**:

Consistent naming.

**How to reproduce it**:

* Create a new load balancer and check the naming.
* Update the load balancer and check the naming again.

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version: Current master (3fbdeed798daf29e5cdcd2a6a6ac0c0bf13794f6)
- OpenStack version:
- Others:

### PR DESCRIPTION
**Pull request by shaardie**
**What this PR does / why we need it**:

This patch introduces names for listener, pools and monitors during creating in
`createFullyPopulatedOctaviaLoadBalancer`.

Those names are currently missing or different to the names during updates.

**Which issue this PR fixes(if applicable)**:
fixes #1965

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/openstack/loadbalancer.go [hunk 1]
```diff
@@ -621,19 +621,21 @@ func (lbaas *LbaasV2) createFullyPopulatedOctaviaLoadBalancer(name, clusterName
 		createOpts.VipAddress = loadBalancerIP
 	}
 
-	for _, port := range service.Spec.Ports {
+	for portIndex, port := range service.Spec.Ports {
 		listenerCreateOpt := lbaas.buildListenerCreateOpt(port, svcConf)
+		listenerCreateOpt.Name = cutString(fmt.Sprintf("listener_%d_%s", portIndex, name))
```
Comment (jichenjc) [inline pkg/openstack/loadbalancer.go:626]: I think using the convention is name_index instead of index_name?

Comment (shaardie) [reply pkg/openstack/loadbalancer.go:626]: It was in previous releases, but it seems to have changed. See 

* Listener: https://github.com/kubernetes/cloud-provider-openstack/blob/85ed2f274bd96152c5603240a9a40634b7217f65/pkg/openstack/loadbalancer.go#L1908
* Pool: https://github.com/kubernetes/cloud-provider-openstack/blob/85ed2f274bd96152c5603240a9a40634b7217f65/pkg/openstack/loadbalancer.go#L1913
* Monitor: https://github.com/kubernetes/cloud-provider-openstack/blob/85ed2f274bd96152c5603240a9a40634b7217f65/pkg/openstack/loadbalancer.go#L1918

**Comment by jichenjc** [comment]
/lgtm
/approve

ok, so let's reuse previous logic as it's already there

**Pull request merged**